app/models/petition.rb

```diff
@@ -156,6 +155,13 @@ def self.counts_by_state
     counts_by_state
   end
 
+  def publish!
+    self.state = Petition::OPEN_STATE
+    self.open_at = Time.zone.now
+    self.closed_at = AppConfig.petition_duration.months.from_now.end_of_day
```

Comment:
How is the comparison done? If we're doing `<` then we can do what you suggested.